### DISCUSSION TITLE
bucket directories size limits

### DISCUSSION THREAD

**Discussion by LinTechSo** [minio/minio#14467] [Q&A] [answered]
HI, Is there a way to set quotas for directories inside each bucket in a MinIO server and monitor the size and quota with the API of each directory in the bucket?

**Comment by ureykane** [comment]
Since object storage adopts a distributed storage method, objects are not physically stored according to folders. That is to say, not all files in a folder will be stored together. In the process of backend storage, the files in different folders are only different in the prefix of the key value. Therefore, under this architecture, it is impossible to conveniently count summary information under a certain folder, such as folder size, number of files, etc.

**Comment by klauspost** [comment, marked as answer]
https://docs.min.io/minio/baremetal/reference/minio-mc-admin/mc-admin-bucket-quota.html#mc-admin-bucket-quota

`mc du` provides usage info.

If you are looking to implement a commercial service you can write `sales@minio.io`, and we can go into details.

**Reply by LinTechSo** [reply]
Thanks. 
Yes, I found this link about the mc admin bucket quota. but this solution is just for buckets not each directory under the bucket.

**Comment by LinTechSo** [comment]
https://docs.min.io/minio/baremetal/reference/minio-mc-admin/mc-admin-bucket-quota.html#mc-admin-bucket-quota

I found this link about the mc admin bucket quota. but this solution is just for buckets not each directory under the bucket.

**Reply by harshavardhana** [reply]
That is out of scope @LinTechSo
